sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/Data.java

```diff
@@ -374,8 +374,23 @@ public void run() {
         }
     }
 
+    @WorkerThread
     private synchronized void processPendingOperations() {
-        for (LocalDocument localDocument : mLocalDocumentStorage.getPendingOperations(Utils.getUserTableName())) {
+        String table = Utils.getUserTableName();
+        for (LocalDocument localDocument : mLocalDocumentStorage.getPendingOperations(table)) {
+            if (localDocument.isExpired()) {
+                if (localDocument.getOperation().equals(Constants.PENDING_OPERATION_PROCESS_VALUE)) {
```

Comment:
can we test that if first and not nest it under `if (localDocument.isExpired())`? It's very confusing otherwise.